### PR TITLE
feat: update provider and module version to latest

### PR DESCRIPTION
**Pull request by gnmeyer**
Update the versions of the "vpc" and "ec2_instance" modules to their latest. Updated the aws_provider version to latest and required_terraform version to latest. Reason: To fix errors depreciation errors in "vpc" module

### CODE REVIEW

**Comment by jburns24** [comment]
@gnmeyer Have you tested these locally?

File: docs/7-infrastructure-configuration-management/7.1.3-terraform-modules.md [hunk 1]
```diff
@@ -117,12 +138,12 @@ module "vpc" {
...
 module "ec2_instance" {
   source  = "terraform-aws-modules/ec2-instance/aws"
-  version = "4.3.0"
+  version = "5.6.1"
 
   count = 1
   name  = "my-ec2-cluster-${count.index}"
 
-  ami                    = "ami-0c5204531f799e0c6"
+  ami                    = "ami-0fe630eb857a6ec83"
```
Comment (jburns24) [inline docs/7-infrastructure-configuration-management/7.1.3-terraform-modules.md:146]: What AMI did you change to? Would be nice to add a comment after the ami to let future maintainers know. Something like `# Amazon Linux 2`

Comment (gnmeyer) [reply docs/7-infrastructure-configuration-management/7.1.3-terraform-modules.md]: Got it, just updated in the most recent commit # Red Hat Enterprise Linux 9

**Comment by gnmeyer** [comment]
Just verified changes work on local

**Review by jburns24 — APPROVED** [review]

**Pull request merged**